src/main/java/seedu/address/model/assignment/DueDate.java

```diff
@@ -62,6 +68,25 @@ public DueDate(String date, String time) {
     public DueDate(String date) {
         this(date, LATEST_TIME_IN_DAY);
     }
+
+    /**
+     * Constructs a {@code DueDate}
+     *
+     * @param friendlyDate friendly command format of date
+     * @param code code value assigned to denote friendly command
+     * @throws ParseException erroneous is given, exception is thrown
+     */
+    public DueDate(String friendlyDate, int code) throws ParseException {
+        if (code == 1) {
+            this.date = friendlyToDate(friendlyDate);
+            this.time = LocalTime.parse(LATEST_TIME_IN_DAY, PARSE_TIME_FORMAT);
+            this.dateTime = LocalDateTime.of(this.date, this.time);
+            this.value = this.dateTime.format(OUTPUT_FORMAT);
+        } else {
+            throw new ParseException(DueDate.MESSAGE_CONSTRAINTS_DUE_DATE);
+        }
+    }
+
```

Comment:
Maybe have the check isFriendlyDate be placed here, so you do need to have separate constructors and use of the FRIENDLY_CODE variable

Reply:
I agree with this, great idea!